src/tests/conftest.py

```diff
@@ -1,23 +1,93 @@
 import pytest
 from starlette.testclient import TestClient
-from datetime import datetime as dt
+from datetime import datetime
 
 from app.main import app
 from app.api.schemas import UserRead, DeviceOut
 from app.api.deps import get_current_user, get_current_device
 
 
 async def mock_current_user():
-    return UserRead(id=99, username="connected_user", created_at=dt.now())
+    return UserRead(id=99, username="connected_user", created_at=datetime.now())
 
 
 async def mock_current_device():
-    return DeviceOut(id=99, owner_id=1, specs="raspberry", name="connected_device", created_at=dt.now())
+    return DeviceOut(id=99, owner_id=1, specs="raspberry", name="connected_device", created_at=datetime.now())
+
+
+# Sterilize all DB interactions
+async def mock_get(entry_id, table):
+    for entry in table:
+        if entry['id'] == entry_id:
+            return entry
+    return None
+
+
+async def mock_fetch_all(table, query_filters=None):
+    if query_filters is None:
+        return table
+    response = []
+    for entry in table:
+        if all(entry[k] == v for k, v in query_filters):
+            response.append(entry)
+    return response
+
+
+async def mock_fetch_one(table, query_filters=None):
+    for entry in table:
+        if all(entry[k] == v for k, v in query_filters):
+            return entry
+    return None
+
+
+async def mock_post(payload, table):
+    payload_dict = payload.dict()
+    payload_dict['created_at'] = datetime.utcnow()
+    payload_dict['id'] = len(table) + 1
+    table.append(payload_dict)
+    return payload_dict['id']
+
+
+async def mock_put(entry_id, payload, table):
+    for idx, entry in enumerate(table):
+        if entry['id'] == entry_id:
+            for k, v in payload.dict().items():
+                table[idx][k] = v
+    return entry_id
+
+
+async def mock_delete(entry_id, table):
+    for idx, entry in enumerate(table):
+        if entry['id'] == entry_id:
+            del table[idx]
+            break
+    return entry_id
+
+
+async def mock_hash_password(password):
+    return f"{password}_hashed"
+
+
+async def mock_verify_password(plain_password, hashed_password):
+    return hashed_password == f"{plain_password}_hashed"
+
+
+def pytest_configure():
```

Comment:
Is it a reserved function name that will be called before launching all the tests?

Reply:
Correct, cf. https://docs.pytest.org/en/stable/writing_plugins.html#registering-custom-markers